url/url_test.cpp

```diff
@@ -634,6 +637,105 @@ int main() {
         etest::expect_eq(*url, url::Url{.scheme = "a", .host = url::Host{.type = url::HostType::Opaque}});
     });
 
+    etest::test("Web Platform Tests", [] {
+        url::UrlParser p;
+
+        simdjson::ondemand::parser parser;
+
+        auto json = simdjson::padded_string::load("external/wpt/url/resources/urltestdata.json");
+
+        simdjson::ondemand::document doc = parser.iterate(json);
+
+        simdjson::ondemand::array arr = doc.get_array();
+
+        for (auto obj : arr) {
+            // Skip strings, those are just comments
+            if (obj.type() == simdjson::ondemand::json_type::string) {
+                continue;
+            }
+
+            bool failure = false;
```

Comment:
Maybe rename this to `should_fail` so it's clearer what it means in this context?